### PR TITLE
trunk action needs test coverage for checkservice callpath

### PR DESCRIPTION
**Pull request by jennygzma**

### CODE REVIEW

File: action_tests/trunk_shim.sh
Comment (sxlijin) [inline action_tests/trunk_shim.sh:1]: let's use the stub.js / assert.js pattern instead - that feels more bulletproof than this

File: .github/workflows/action_test.yaml [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+      - name: Craft TEST_GITHUB_EVENT_PATH
+        shell: bash
+        # GITHUB_EVENT_PATH is the env var of path to file on runner that contains full webhook payload.
+        # we are replacing it with TEST_GITHUB_EVENT_PATH so we read our predefined payload for test cases.
+        run: |
+          TEST_GITHUB_EVENT_PATH=$(mktemp)
+          cat >> $TEST_GITHUB_EVENT_PATH <<EOF
+          ${{matrix.payload}}
+          EOF
+          echo "TEST_GITHUB_EVENT_PATH=TEST_GITHUB_EVENT_PATH" >>$GITHUB_ENV
```
Comment (sxlijin) [inline .github/workflows/action_test.yaml:93]: looking good! obvs this needs to become `echo TGEP=$TGEP >>env`

---

in addition, you could also switch to doing `TEST_GITHUB_EVENT_PATH=${GITHUB_ACTION_PATH}/action_tests/trunk_merge_test_payload.json`

---

to take it a step further, you can also do something like

```
TEST_GITHUB_EVENT_PATH=$(mktemp)
echo "{ \"inputs\": { \"payload\": $(cat ${{ matrix.payload }} | jq 'tojson') } }" >>$TEST_GITHUB_EVENT_PATH
echo TEST_GITHUB_EVENT_PATH=$TEST_GITHUB_EVENT_PATH >>$GITHUB_ENV
```

 and then all you would have to do would be to shove the contents of `inputs.payload` into the path in `matrix.payload`

File: action.yaml [hunk 2]
```diff
@@ -193,11 +194,10 @@ runs:
         INPUT_UPLOAD_SERIES=${{ inputs.upload-series }}
         INPUT_UPLOAD_ID=
         EOF
-
         fi
 
     - name: Checkout
-      if: env.INPUT_TARGET_CHECKOUT
+      if: env.INPUT_TARGET_CHECKOUT && TEST_GITHUB_EVENT_PATH == ""
```
Comment (jennygzma) [inline action.yaml:200]: @sxlijin is this what u mean by omitting this step? still seems to fail

Comment (sxlijin) [reply action.yaml:200]: just don't specify `targetCheckout` in your payload

Comment (jennygzma) [reply action.yaml]: hm i think im still struggling. -do u mind taking a look? sorry its kinda messy but just within action_tests.yml

**Comment by jennygzma** [comment]
too messy, handling here: https://github.com/trunk-io/trunk-action/pull/165

**Pull request closed without merging**